### DISCUSSION TITLE
Adding different Start and End nodes to capacitated VRP with Pickup&Delivery with Time Windows

### DISCUSSION THREAD

**Discussion by Lan61** [google/or-tools#3624] [Routing (and legacy CP) questions]
I want to extend the CVRPPDTW from: https://gist.github.com/Mizux/4870e4af09b31dd45b7ac632d1830a73

1.	Is it possible to have node 0 (depot) as a starting point and 1 (another depot) as ending point? What exactly do I have to adjust?
2.	How do I add different service times to each customer node? I know the problem is related to the one from: https://github.com/google/or-tools/blob/stable/ortools/constraint_solver/samples/cvrptw.py#L162, but I am not able to tackle the problem so far. 

I am a beginner programmer and I am a bit confused how I should manage it.

**Comment by Mizux** [comment]
1. Simply pass an array of start and end nodes, size of the array is equal to the number of vehicle
see: https://developers.google.com/optimization/routing/routing_tasks#setting-start-and-end-locations-for-routes

2. How do you map your "customer" ?  In the model, are they the "node" to visit or the "vehicle" ?
